src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,123 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+            this.next = null;
+        }
+    }
+
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size = 0;
+    private int capacity;
+    private final float loadFactor;
+    private V nullKeyValue;
+
+    public MyHashMap() {
+        this.capacity = INITIAL_CAPACITY;
+        this.loadFactor = DEFAULT_LOAD_FACTOR;
+        this.table = new Node[capacity];
+
+    }
 
     @Override
     public void put(K key, V value) {
 
+        if (key == null) {
+            putNullKey(value);
+            return;
+        }
+
+        int index = getIndex(key);
+        Node<K,V> newNode = new Node<>(key,value);
```

Comment:
The Node class should be parameterized to prevent problems with data types. Use 'Node<K, V>' instead of just 'Node'.